src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/rc_pending_resumption_handler.h

```diff
@@ -0,0 +1,173 @@
+#include "application_manager/rpc_service.h"
+#include "rc_rpc_plugin/interior_data_cache.h"
+#include "rc_rpc_plugin/rc_app_extension.h"
+
+namespace rc_rpc_plugin {
+
+/**
+ * @brief The RCPendingResumptionHandler class
+ * responsibility to avoid duplication of subscription requests to HMI
+ * if multiple applications are registering
+ */
+class RCPendingResumptionHandler
+    : public resumption::ExtensionPendingResumptionHandler {
+ public:
+  RCPendingResumptionHandler(
+      application_manager::ApplicationManager& application_manager,
+      rc_rpc_plugin::InteriorDataCache& interior_data_cache);
+
+  void on_event(const application_manager::event_engine::Event& event) override;
+
+  void HandleResumptionSubscriptionRequest(
+      application_manager::AppExtension& extension,
+      resumption::Subscriber& subscriber,
+      application_manager::Application& app) override;
+
+  void OnResumptionRevert() override;
+
+  /**
+   * @brief Creates GetInteriorVehicleData
+   * subscription request
+   * @param module unique identify a module
+   * moduleType + moduleID
+   * @param correlation_id - unique ID
+   * @return subscription request in smart object representation
+   */
+  static smart_objects::SmartObjectSPtr CreateSubscriptionRequest(
+      const ModuleUid& module, const uint32_t correlation_id);
+
+  /**
+   * @brief Retrieves function id from
+   * subscription request
+   * @param subscription_request a subscription request that contains the
+   * function id
+   * @return function id
+   */
+  static hmi_apis::FunctionID::eType GetFunctionId(
+      const smart_objects::SmartObject& subscription_request);
+
+  /**
+   * @brief Retrieves module uid from
+   * subscription request
+   * @param subscription_request a subscription request that contains a unique
+   * identify a module
+   * @return unique identify a module
+   */
+  static ModuleUid GetModuleUid(
+      const smart_objects::SmartObject& subscription_request);
+
+ private:
+  /**
+   * @brief The PendingRequest struct contains fields, needed during
+   * processing events, related to responses from HMI to each resumption request
+   */
+  struct PendingRequest {
+    uint32_t app_id;
+    smart_objects::SmartObject message;
+    uint32_t correlation_id() const {
+      namespace am_strings = app_mngr::strings;
+      return message[am_strings::params][am_strings::correlation_id].asInt();
+    }
+  };
+
+  /**
+   * @brief Handles a successful response from HMI
+   * @param event contains response from HMI
+   * @param module_uid a unique identify a module
+   */
+  void HandleSuccessfulResponse(
+      const application_manager::event_engine::Event& event,
+      const ModuleUid& module_uid);
+
+  /**
+   * @brief Creates next GetInteriorVehicleData
+   * subscription request if a previous resumption of subscriptions is not
+   * successful
+   * @param module a unique identify a module
+   */
+  void ProcessNextFreezedResumption(const ModuleUid& module);
```

Comment:
```suggestion
  void ProcessNextPausedResumption(const ModuleUid& module);
```
Still would prefer a name change from `Freezed`, seems `Paused` would work fine

Reply:
@jacobkeeler Done 07385665dccf8609db73e02f4cacabe73fa699e5